driftbase/utils/test_utils.py

```diff
@@ -82,6 +89,27 @@ def _create_server(self, machine_id):
         resp = self.post("/servers", data=data, expected_status_code=http_client.CREATED)
         return resp.json()
 
+    @contextmanager
+    def _app_context(self):
+        with self.drift_app.app_context() as ctx:
+            drift_config = self.drift_app.extensions['driftconfig']
+            ctx.driftconfig = get_config()
+            yield ctx
+
+    @contextmanager
+    def _request_context(self, *args, **kwargs):
+        with self._app_context():
+            if "path" not in kwargs:
+                kwargs["path"] = "/test/endpoint"
+            with self.drift_app.test_request_context(*args, **kwargs) as request_ctx:
+                g.conf = LocalProxy(get_config_for_request)
+                g.db = LocalProxy(get_sqlalchemy_session)
+                g.redis = LocalProxy(get_redis_session)
+                try:
+                    yield request_ctx
+                finally:
+                    gevent.idle()
+
```

Comment:
You don't need these; all your tests go through an api, so you don't need to fake going through an api.

Reply:
I need it to call `rp.set_richpresence(friend_id, presence)` in one of the tests. I will revisit this topic after refactoring the service to be better encapsulated from the flask extension.

Reply:
Note: These context managers already existed. I just moved them "up" one level so they're available in the test utils and usable across all tests, rather than locked to the shoutout implementation.

Reply:
We should probably move these further up to drift's systesthelper at some point